backend/src/tasks/dnstwist.ts

```diff
@@ -30,13 +28,33 @@ async function runDNSTwist(domain: Domain) {
 export const handler = async (commandOptions: CommandOptions) => {
   const { organizationId, organizationName } = commandOptions;
 
+  await connectToDatabase;
+  const date = new Date(Date.now());
   console.log('Running dnstwist on organization', organizationName);
   const domainsWithIPs = await getIps(organizationId);
 
   const vulns: Vulnerability[] = [];
   for (const domain of domainsWithIPs) {
     try {
       const results = await runDNSTwist(domain);
+      // Fetch existing dnstwist vulnerability
+      const existingVuln = await Vulnerability.findOne({
+        domain: { id: domain.id },
+        source: 'dnstwist'
+      });
+      for (const domain of results) {
+        domain['date-observed'] = date;
+        if (existingVuln) {
+          // If domain in existingVuln, keep the existing date-observed
+          existingVuln.structuredData['domains'].map(
```

Comment:
Use forEach instead of map because you don't need to do anything with the function results.

```suggestion
          existingVuln.structuredData['domains'].forEach(
```